cli/azd/pkg/containerapps/container_app.go

```diff
@@ -146,6 +146,7 @@ func (cas *containerAppService) persistSettings(
 	aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName, options)
 	if err != nil {
 		log.Printf("failed getting current aca settings: %v. No settings will be persisted.", err)
+		return obj, nil
```

Comment:
This is slightly better and probably fixes the regression... But I'm wondering if this also opens us up to weird future problems that we want to avoid entirely?

Imagine if we issued a `GET` request that results in 503 Server Unavailable, we end up not persisting the settings here, whereas I could imagine the right behavior is to fail, since we can't evaluate the diff?

I'm unsure about the background here behind the "soft failure" here -- are we trying to express that a 404 should be "no changes necessary here?"

Reply:
I'm in agreement with @weikanglim - my understanding of this function is that it's trying to merge a subset of properties from the existing container app (if it exists) with the desired state expressed by the YAML because we want to allow certain out of band edits to be applied over the generated desired state (to save folks the trouble of `infra synth`'ing and then editing the yaml themselves).  In Aspire 9 this is no longer needed because the customer can edit these properties directly in the app host (and if they wanted do similar logic to what we have in in C# and decide how they want to handle each failure).

For our case I feel like we should treat 404 as "we're creating this app so there's nothing to merge into the desired state" and other errors as "we can't be sure what's going on so we should conservatively fail the update with an error message".

I think that `persistSettings` is sort of a bad name at this point, maybe `mergePersistedSettingsFromExistingApp` or something to it's a little clearer from the function name what exactly this is intended to do?

Reply:
I thought about creating a unique error for `Not Found` . But this is an alpha feature which is planned to be retired/deprecated soon. 
Aspire already provides in version 9 a way to model the ACA settings directly in the AppHost and there is no need to ask azd to persist previous settings.
Actually, when moving to Aspire9 + `PublishAsAzureContainerApp()` like here: https://github.com/dotnet/aspire/blob/main/playground/AzureContainerApps/AzureContainerApps.AppHost/Program.cs#L29C9-L29C35 the `persisting feature` from azd becomes ignored/no-op.

We want anyone currently depending on persist settings with azd to use Aspire9.

That made me not wanting to invest more here into a better implementation for this.

Reply:
If we do go down this line then I could image a helpful comment above the `return obj, nil` along the lines of "if the container app doesn't exist, there's nothing for us to update in the desired state, so we can just return the existing state as is."